src/arch/riscv/se_workload.hh

```diff
@@ -61,14 +62,33 @@ class SEWorkload : public gem5::SEWorkload
     loader::Arch getArch() const override { return loader::Riscv64; }
 
     using SyscallABI64 = RegABI64;
-    using SyscallABI32 = RegABI32;
+
+    struct SyscallABI32 : public GenericSyscallABI32
+    {
+        static const std::vector<RegId> ArgumentRegs;
+    };
+
 };
 
 } // namespace RiscvISA
 
 namespace guest_abi
 {
 
+template<typename ABI, typename Arg>
+struct Argument<ABI, Arg,
+    std::enable_if_t<
+        std::is_base_of_v<ABI, RiscvISA::SEWorkload::SyscallABI32> &&
+        std::is_integral_v<Arg> &&
+        ABI::template IsWideV<Arg>>>
+{
+    static Arg
+    get(ThreadContext *tc, typename ABI::State &state)
+    {
+        return (Arg) bits(tc->getReg(ABI::ArgumentRegs[state++]), 31, 0);
+    }
+};
+
```

Comment:
Sorry I need to request you to change the RISC-V system ABI call. I think it's not a good way to handle the `uint64_t` and `int64_t` in RV32 SE mode.

I have check the type of `size_t` and `off_t` in ARM and X86 Linux Call

ARMLinux32
https://github.com/gem5/gem5/blob/0ec8cf8d05bb76b16d41d799d99323867c50c885/src/arch/arm/linux/linux.hh#L115-L120

X86Linux32
https://github.com/gem5/gem5/blob/0ec8cf8d05bb76b16d41d799d99323867c50c885/src/arch/x86/linux/linux.hh#L345-L349

Can we just declare the size_t and off_t as `uint32_t` in 'RiscvLinux32` instead of change `RiscvISA::SEWorkload::SyscallABI32`?